app/lib/handleKeySigInteraction.ts

```diff
@@ -1,28 +1,46 @@
 import {
   removeAccidentalFromNotesAndCoords,
-  updateNotesAndCoordsWithAccidental,
+  updateNotesAndCoordsWithAccidentalForKeySig,
 } from "../lib/modifyNotesAndCoordinates";
-import { KeySigState, NotesAndCoordinatesData } from "./typesAndInterfaces";
+import {
+  addGlyphs,
+  deleteAccidentalFromKeySigArray,
+  deleteGlyphFromStave,
+  updateKeySigArrayForGrading,
+} from "./modifyKeySignature";
+import {
+  GlyphProps,
+  KeySigState,
+  NotesAndCoordinatesData,
+} from "./typesAndInterfaces";
 
 export const handleKeySigInteraction = (
   notesAndCoordinates: NotesAndCoordinatesData[],
   state: KeySigState,
-  foundNoteData: NotesAndCoordinatesData
-  //keySig: string[],
+  foundNoteData: NotesAndCoordinatesData,
+  xClick: number,
+  yClick: number,
+  glyphState: (newState: React.SetStateAction<GlyphProps[]>) => void,
+  keySigState: (newState: React.SetStateAction<string[]>) => void,
+  keySig: string[]
 ) => {
-  if (state.isSharpActive || state.isFlatActive) {
-    notesAndCoordinates = updateNotesAndCoordsWithAccidental(
+  if (state.isAddSharpActive || state.isAddFlatActive) {
+    notesAndCoordinates = updateNotesAndCoordsWithAccidentalForKeySig(
       state,
       foundNoteData,
       notesAndCoordinates
     );
-  } else if (state.isEraseAccidentalActive) {
-    notesAndCoordinates = removeAccidentalFromNotesAndCoords(
-      notesAndCoordinates,
-      foundNoteData
-    );
-  } else {
-  }
+    addGlyphs(xClick, yClick, state, glyphState);
+    updateKeySigArrayForGrading(foundNoteData, state, keySigState);
+  } 
+  // else if (state.isRemoveAccidentalActive) {
```

Comment:
Does this need to be deleted?

Reply:
i can't delete that yet because that is the feature I am having trouble with. It is part of the "delete accidental" feature. Worst case scenario, we don't have that feature. The most important thing is that the accidentals are accurately tracked.